File: learn_java/baseAOC.java
import java.nio.file.*;
import java.io.IOException;
import java.util.List;

public class FileProcessor {

    public static void main(String[] args) {
        try {
            // 1. Cargar todas las líneas del fichero
            List<String> lines = Files.readAllLines(Path.of("input.txt"));

            // 2. Procesar las líneas para obtener un resultado numérico
            int result = processLines(lines);

            // 3. Mostrar el resultado numérico por pantalla
            System.out.println("Resultado del procesamiento: " + result);

        } catch (IOException e) {
            System.out.println("Error leyendo el fichero: " + e.getMessage());
        }
    }

    /**
     * Procesa las líneas del fichero y devuelve un resultado numérico.
     */
    private static int processLines(List<String> lines) {
         List<Integer> left = new ArrayList<>();
        List<Integer> right = new ArrayList<>();

        // Parsear cada línea en dos columnas
        for (String line : lines) {
            if (line.isBlank()) {
                continue; // por si hay líneas vacías
            }

            // Dividir por espacios (uno o más)
            String[] parts = line.trim().split("\\s+");
            int leftValue = Integer.parseInt(parts[0]);
            int rightValue = Integer.parseInt(parts[1]);

            left.add(leftValue);
            right.add(rightValue);
        }

        // Ordenar ambas listas
        Collections.sort(left);
        Collections.sort(right);

        // Sumar las distancias absolutas
        long result = 0L;
        for (int i = 0; i < left.size(); i++) {
            int a = left.get(i);
            int b = right.get(i);
            result += Math.abs(a - b);
        }

        return result;
    }
}

File: learn_rust/src/bin/baseAOC.rs
use std::fs;
use std::io;

fn main() -> io::Result<()> {
    // 1. Leer todo el fichero como String
    let content = fs::read_to_string("input.txt")?;

    // 2. Convertir en líneas
    let lines: Vec<&str> = content.lines().collect();

    // 3. Procesar líneas y obtener un número
    let result = process_lines(&lines);

    // 4. Mostrar resultado
    println!("Resultado del procesamiento: {}", result);

    Ok(())
}

/// Procesa las líneas y devuelve un resultado numérico.
/// Aquí es donde luego metes la lógica que quieras (TBD).
fn process_lines(lines: &[&str]) -> i64 {
    let mut result: i64 = 0;

    for line in lines {
        let line = line.trim();
        if line.is_empty() {
            continue;
        }

        // TODO: lógica concreta
        // Ejemplo de plantilla:
        // if let Ok(value) = line.parse::<i64>() {
        //     result += value;
        // }

    }

    result
}